### PR TITLE
Glossary

### PR DESCRIPTION
**Pull request by OndroMih**
**!!! Please do not merge before discussed properly on the PR or on the mailing list !!!**

As suggested in the mailing list: https://groups.google.com/d/msg/microprofile/j9urpwSd6es/JKoDdnLNEAAJ

This is only a suggestion, so that it can be commented and updated if agreement is established.

### CODE REVIEW

File: glossary.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+= Glossary of common terms within MicroProfile
+
+[options="header"]
+|===
+| Term | Description
+| Evolution process | A formal process to evolve what is included the MicroProfile
+| Proposal | A proposal for a new addition to the MicroProfile (a formal document describing what should be added and why)
+| Candidate | The work in process after a new proposal within the evolution process was accepted, and before the proposal comes to a final version so that it can become part of MicroProfile. May also refer to the repository related to an accepted proposal. Should include the definition of the proposed API or behavior. May also include a reference implementation and a compatibility kit
```
Comment (OndrejM) [inline glossary.adoc]: Other possible terms: prototype, concept

Comment (OndrejM) [inline glossary.adoc]: @kenfinnigan proposed in the [mailing list](https://groups.google.com/d/msg/microprofile/j9urpwSd6es/YdSLbZDbEAAJ):

> I would lean towards "concept", as "candidate" can often have meanings associated with a near release. ie. release candidate.

File: glossary.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+| Specification |  The final delivery of a "candidate" (see above), is a result of an accepted proposal, usually specifies an API or standard functionality,  may be included in MicroProfile
+| Implementation | In a similar sense as a JSR implementation - implementation of a certain API or behavior prescribed by a specification
+| Reference implementation | In a similar sense as a JSR reference implementation - an implementation provided and maintained by the MicroProfile project itself
+| MicroProfile implementation | An implementation of all API and behavior prescribed MicroProfile
+| MicroProfile Compatibility Kit (MCK) | compatibility suite/kit, which tests that an implementation is compatible with a specification or complete MicroProfile
+| Reference applications | applications maitained by the MicroProfile project to demonstrate usage of MicroProfile to develop a complete application or system architecture
+| MP | A shorthand for MicroProfile
```
Comment (otaviojava) [inline glossary.adoc]: @OndrejM 
One important point is about the lifecycle of a proposal.
:question: How long a proposal can be worked? One year? Two year?

Even an easier one such as:

* Incubation: the initial process to start 
* WIP
* Release

File: glossary.adoc (hunk 1 above)
Comment (OndrejM) [inline glossary.adoc]: I like "concept" too. Another word that comes to mind: "prototype".

File: glossary.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+| Candidate | The work in process after a new proposal within the evolution process was accepted, and before the proposal comes to a final version so that it can become part of MicroProfile. May also refer to the repository related to an accepted proposal. Should include the definition of the proposed API or behavior. May also include a reference implementation and a compatibility kit
+| Specification |  The final delivery of a "candidate" (see above), is a result of an accepted proposal, usually specifies an API or standard functionality,  may be included in MicroProfile
+| Implementation | In a similar sense as a JSR implementation - implementation of a certain API or behavior prescribed by a specification
+| Reference implementation | In a similar sense as a JSR reference implementation - an implementation provided and maintained by the MicroProfile project itself
+| MicroProfile implementation | An implementation of all API and behavior prescribed MicroProfile
+| MicroProfile Compatibility Kit (MCK) | compatibility suite/kit, which tests that an implementation is compatible with a specification or complete MicroProfile
```
Comment (OndrejM) [inline glossary.adoc]: We could use Technology Compatibility Kit, but as it is very often coupled with JCP and JSRs (https://en.wikipedia.org/wiki/Technology_Compatibility_Kit), as suggested in the mailing list, I recommend to bind to the term "MicroProfile Compatibility Kit" (MCK).

File: glossary.adoc (hunk 1 above)
Comment (emilyjiang) [inline glossary.adoc]: +1 on prototype

File: glossary.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+[options="header"]
+|===
+| Term | Description
+| Evolution process | A formal process to evolve what is included the MicroProfile
+| Proposal | A proposal for a new addition to the MicroProfile (a formal document describing what should be added and why)
+| Candidate | The work in process after a new proposal within the evolution process was accepted, and before the proposal comes to a final version so that it can become part of MicroProfile. May also refer to the repository related to an accepted proposal. Should include the definition of the proposed API or behavior. May also include a reference implementation and a compatibility kit
+| Specification |  The final delivery of a "candidate" (see above), is a result of an accepted proposal, usually specifies an API or standard functionality,  may be included in MicroProfile
+| Implementation | In a similar sense as a JSR implementation - implementation of a certain API or behavior prescribed by a specification
+| Reference implementation | In a similar sense as a JSR reference implementation - an implementation provided and maintained by the MicroProfile project itself
```
Comment (OndrejM) [inline glossary.adoc]: We've decided that we won't deliver a reference implementation. Each specification should be released with at least one implementation, that passes the TCK, with no other requirements.

File: glossary.adoc (hunk 3 above)
Comment (OndrejM) [inline glossary.adoc]: The term TCK got adopted in the configuration spec and onbody objected, therefore we should probably redefine this to MicroProfile Technology Compatibility Kit (MicroProfile TCK, or just TCK).

**Comment by kwsutter** [comment]
Since there has been no activity with this PR for almost 2 years, I'm going to close it.  If there is strong desire to still merge this content, re-open and re-fresh the PR.  Thanks!

**Pull request closed without merging**